components/script/dom/performance.rs

```diff
@@ -273,10 +287,70 @@ impl Performance {
             o.notify();
         }
     }
-
     fn now(&self) -> f64 {
         (time::precise_time_ns() - self.navigation_start_precise).to_ms()
     }
+    fn can_add_resource_timing_entry(&self) -> bool {
+        self.resource_timing_buffer_current_size.get() <=
+            self.resource_timing_buffer_size_limit.get()
+    }
+    fn copy_secondary_resource_timing_buffer(&self) {
+        while self.can_add_resource_timing_entry() && !self
+            .resource_timing_secondary_entries
+            .borrow_mut()
+            .is_empty()
+        {
+            self.queue_entry(
+                &(*(self
+                    .resource_timing_secondary_entries
+                    .borrow_mut()
+                    .pop_front()
+                    .unwrap())),
+                true,
+            );
+        }
+    }
+    fn fire_buffer_full_event(&self) {
+        while !self
+            .resource_timing_secondary_entries
+            .borrow_mut()
+            .is_empty()
+        {
+            let no_of_excess_entries_before =
+                self.resource_timing_secondary_entries.borrow_mut().len();
+            let mut no_of_excess_entries_after =
+                self.resource_timing_secondary_entries.borrow_mut().len();
+            if !self.can_add_resource_timing_entry() {
+                self.eventtarget
+                    .fire_event(atom!("resourcetimingbufferfull"));
+                self.copy_secondary_resource_timing_buffer();
+                no_of_excess_entries_after =
+                    self.resource_timing_secondary_entries.borrow_mut().len();
+            }
+            if no_of_excess_entries_after >= no_of_excess_entries_before {
+                self.resource_timing_secondary_entries.borrow_mut().clear();
+                break;
+            }
+        }
+        self.resource_timing_buffer_pending_full_event.set(false);
+    }
+    fn should_queue_resource_entry(&self, entry: &PerformanceEntry) -> bool {
+        if self.can_add_resource_timing_entry() &&
+            !self.resource_timing_buffer_pending_full_event.get()
+        {
```

Comment:
In this case we should only increase `resource_timing_buffer_current_size` and return true, so the resource entry is queued by the `queue_entry` function. Otherwise, we should implement the rest of the steps (3, 4 and 5 of the spec) here before returning `false`.